trek.js

```diff
@@ -0,0 +1,131 @@
+const listErrors = (errors) => {
+  let errorList = '<ul>';
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      errorList += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  errorList += '</ul>';
+  return errorList;
+};
+
+const showMessage = (elementID, status) => {
+  $(elementID).addClass(status);
+  $(elementID).removeClass('hidden');
+};
+
+const handleReserveTrip = (tripID, continent) => {
+  const endpoint = `https://trektravel.herokuapp.com/trips/${tripID}/reservations`;
+  const name = $('#name-field').val();
+  const email = $('#email-field').val();
+
+  axios.post(endpoint, { name, email,})
+    .then((response) => {
+      $('#reserve-form')[0].reset();
+      showMessage('#reserve-message','success');
+
+      const messageHTML = (
+        `<h4>Congrats, ${name}, you're going to ${continent}!</h4>`
+              + `<p>Your reservation ID is ${response.data.id}.</p>`
+      );
+      $('#reserve-message').html(messageHTML);
+      
+    })
+    .catch((error) => {
+      showMessage('#reserve-message','error');
+      $('#reserve-message').html(`<h4> A problem occured: ${error.message} </h4>`);
+
+      if (error.response && error.response.data && error.response.data.errors) {
+        const errors = error.response.data.errors;
+        $('#reserve-message').append(listErrors(errors));
+      }
+    });
+};
+
+const changeSelected = (tripID) => {
+  $('.selected').removeClass('selected');
+  $(`#trip-${tripID}`).addClass('selected');
+};
+
+const displayTripDescription = (tripID) => {
+  axios.get(`https://trektravel.herokuapp.com/trips/${tripID}`)
+    .then((response) => {
+      $('#trip-details').append(`<details>${response.data.about}</details>`);
+    })
+    .catch((error) => {
+      $('#trip-details').append(`<p class="error">Failed to load trip details:  ${error.message}</p>`);
+    });
+};
+
+const displayReservationForm = (tripName) => {
+  $('#reserve-heading').text(`Reserve a Spot on ${tripName}`);
+  $('#reserve-message').empty();
+  $('#reserve-trip').removeClass('hidden');
+};
+
+const setReservationButtonHandler = (tripID, continent) => {
+  $('#reserve-button').off();
+  $('#reserve-button').click((event) => {
+    event.preventDefault();
```

Comment:
Instead of a `click` on the button, you should listen for a `submit` event on the form itself. This will fire both when the button is clicked and when the user presses `<enter>`.

Also, good catch turning off any previous event handler - this avoids a tricky bug with sending multiple POST requests.